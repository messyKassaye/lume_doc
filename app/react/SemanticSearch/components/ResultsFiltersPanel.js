import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Form } from 'react-redux-form';
import { NumericRangeSlide } from 'app/ReactReduxForms';
import { Icon } from 'UI';

import SidePanel from 'app/Layout/SidePanel';
import { t, Translate } from 'app/I18N';

const filters = [
  {
    label: 'Threshold',
    model: 'threshold',
    min: 0.3,
    max: 1,
    step: 0.01,
    minLabel: 'Exploration',
    maxLabel: 'Precision',
  },
  {
    label: 'Minimum relevant sentences per document',
    model: 'minRelevantSentences',
    min: 1,
    max: 50,
    step: 1,
    minLabel: '',
    maxLabel: '',
  },
];

const filterValue = (filter, filtersValues) =>
  filter.model === 'threshold'
    ? `${(filtersValues[filter.model] * 100).toFixed(2)}%`
    : filtersValues[filter.model];

export function ResultsFiltersPanel({ open, filtersValues }) {
  return (
    <SidePanel open={open}>
      <div className="sidepanel-body">
        <div className="sidepanel-title">{t('System', 'Fine tune')}</div>
        <Form model="semanticSearch.resultsFilters">
          <div className="view">
            {filters.map(filter => (
              <dl className="metadata-type-text" key={filter.label}>
                <dt>
                  {t('System', filter.label)} {filterValue(filter, filtersValues)}
                </dt>
                <dd>
                  <NumericRangeSlide
                    delay={200}
                    model={`.${filter.model}`}
                    prefix={filter.model}
                    min={filter.min}
                    max={filter.max}
                    step={filter.step}
                    minLabel={t('System', filter.minLabel)}
                    maxLabel={t('System', filter.maxLabel)}
                  />
                </dd>
              </dl>
            ))}
          </div>
        </Form>
        <div className="semantic-search-help">
          <div className="alert alert-info alert-vertical">
            <Icon icon="info-circle" size="2x" />
            <p>
              <Translate translationKey="Semantic search overview">
                Semantic search is a technique to provide contextual results. Its ability to capture
                concepts and word associations in human language enables the retrieval of related
                information such as synonyms, connected categories or entities, etc. .
              </Translate>
            </p>
            <p>
              <Translate translationKey="Semantic search threshold help">
                The threshold determines how close the results match the search concept. Move the
                slider to the right to narrow down the concept of the search query. The obtained
                results will be more precise. Move the slider to the left to more broaden the
                concept and explore related content.
              </Translate>
            </p>
            <p>
              <Translate translationKey="Semantic search minimum sentences help">
                Semantic search is applied to each sentence in a document. Filter the documents by
                the minimum number of sentences that exceed the threshold.
              </Translate>
            </p>
          </div>
        </div>
      </div>
    </SidePanel>
  );
}

ResultsFiltersPanel.propTypes = {
  open: PropTypes.bool.isRequired,
  filtersValues: PropTypes.shape({
    threshold: PropTypes.number,
  }).isRequired,
};

function mapStateToProps({ semanticSearch }) {
  return {
    open: semanticSearch.selectedDocument.isEmpty(),
    filtersValues: semanticSearch.resultsFilters,
  };
}

export default connect(mapStateToProps)(ResultsFiltersPanel);
