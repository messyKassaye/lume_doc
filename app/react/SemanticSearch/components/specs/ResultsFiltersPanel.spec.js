import React from 'react';
import { shallow } from 'enzyme';

import { ResultsFiltersPanel } from '../ResultsFiltersPanel';

describe('ResultsFiltersPanel', () => {
  let props;
  let component;
  beforeEach(() => {
    props = {
      open: true,
      filtersValues: { threshold: 0.8 },
    };

    component = shallow(<ResultsFiltersPanel {...props} />);
  });

  describe('render', () => {
    it('should render search filters and instructions', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
