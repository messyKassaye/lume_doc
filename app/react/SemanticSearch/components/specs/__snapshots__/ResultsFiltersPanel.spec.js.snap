// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFiltersPanel render should render search filters and instructions 1`] = `
<SidePanel
  mode=""
  open={true}
>
  <div
    className="sidepanel-body"
  >
    <div
      className="sidepanel-title"
    >
      Fine tune
    </div>
    <Connect(Form)
      model="semanticSearch.resultsFilters"
    >
      <div
        className="view"
      >
        <dl
          className="metadata-type-text"
          key="Threshold"
        >
          <dt>
            Threshold
             
            80.00%
          </dt>
          <dd>
            <NumericRangeSlide
              delay={200}
              max={1}
              maxLabel="Precision"
              min={0.3}
              minLabel="Exploration"
              model=".threshold"
              prefix="threshold"
              step={0.01}
            />
          </dd>
        </dl>
        <dl
          className="metadata-type-text"
          key="Minimum relevant sentences per document"
        >
          <dt>
            Minimum relevant sentences per document
             
          </dt>
          <dd>
            <NumericRangeSlide
              delay={200}
              max={50}
              maxLabel=""
              min={1}
              minLabel=""
              model=".minRelevantSentences"
              prefix="minRelevantSentences"
              step={1}
            />
          </dd>
        </dl>
      </div>
    </Connect(Form)>
    <div
      className="semantic-search-help"
    >
      <div
        className="alert alert-info alert-vertical"
      >
        <Connect(Icon)
          icon="info-circle"
          size="2x"
        />
        <p>
          <Connect(Translate)
            translationKey="Semantic search overview"
          >
            Semantic search is a technique to provide contextual results. Its ability to capture concepts and word associations in human language enables the retrieval of related information such as synonyms, connected categories or entities, etc. .
          </Connect(Translate)>
        </p>
        <p>
          <Connect(Translate)
            translationKey="Semantic search threshold help"
          >
            The threshold determines how close the results match the search concept. Move the slider to the right to narrow down the concept of the search query. The obtained results will be more precise. Move the slider to the left to more broaden the concept and explore related content.
          </Connect(Translate)>
        </p>
        <p>
          <Connect(Translate)
            translationKey="Semantic search minimum sentences help"
          >
            Semantic search is applied to each sentence in a document. Filter the documents by the minimum number of sentences that exceed the threshold.
          </Connect(Translate)>
        </p>
      </div>
    </div>
  </div>
</SidePanel>
`;
